monkeytype/cli.py

```diff
@@ -137,12 +138,36 @@ def apply_stub_handler(args: argparse.Namespace, stdout: IO, stderr: IO) -> None
             raise HandlerError(f"Failed applying stub with retype:\n{cpe.stdout.decode('utf-8')}")
 
 
-def print_stub_handler(args: argparse.Namespace, stdout: IO, stderr: IO) -> None:
+def get_diff(args: argparse.Namespace, stdout: IO, stderr: IO) -> Optional[str]:
+    args.ignore_existing_annotations = False
     stub = get_stub(args, stdout, stderr)
-    if stub is None:
-        print(f'No traces found', file=stderr)
-        return
-    print(stub.render(), file=stdout)
+    args.ignore_existing_annotations = True
+    stub_ignore_anno = get_stub(args, stdout, stderr)
+    if stub is None or stub_ignore_anno is None:
+        return None
+    diff = []
+    seq1 = (s + "\n" for s in stub.render().split("\n\n\n"))
```

Comment:
Are modules/functions guaranteed to appear in the same order between two calls to get stub? If not, the diff may be incorrect.

Reply:
This is a very tricky question.  Because in my added function, I split by `\n\n\n`, because this is how they are joint in `ModuleStub`'s `render` function. 

This would involve `render` in both ` ModuleStub` and `ImportBlockStub`. The `render` in `MonduleStub` do sorting on their names. (full code attached below)
```
        if self.imports_stub.imports:  
            parts.append(self.imports_stub.render())
        for func_stub in sorted(self.function_stubs.values(), key=lambda s: s.name):
            parts.append(func_stub.render())
        for class_stub in sorted(self.class_stubs.values(), key=lambda s: s.name):
            parts.append(class_stub.render())
```
In `render` of `imports_stub` (with type: `ImportBlockStub`), it is also sorted. (full code attached below)
```
        for module in sorted(self.imports.keys()):
            names = sorted(self.imports[module])
```
Full code is listed below.

```python
class ModuleStub(Stub):
    def __init__(
            self,
            function_stubs: Iterable[FunctionStub] = None,
            class_stubs: Iterable[ClassStub] = None,
            imports_stub: ImportBlockStub = None
    ) -> None:
        self.function_stubs: Dict[str, FunctionStub] = {}
        if function_stubs is not None:
            self.function_stubs = {stub.name: stub for stub in function_stubs}
        self.class_stubs: Dict[str, ClassStub] = {}
        if class_stubs is not None:
            self.class_stubs = {stub.name: stub for stub in class_stubs}
        self.imports_stub = imports_stub if imports_stub else ImportBlockStub()

    def render(self) -> str:
        parts = []
        if self.imports_stub.imports:
            parts.append(self.imports_stub.render())
        for func_stub in sorted(self.function_stubs.values(), key=lambda s: s.name):
            parts.append(func_stub.render())
        for class_stub in sorted(self.class_stubs.values(), key=lambda s: s.name):
            parts.append(class_stub.render())
        return "\n\n\n".join(parts)

    def __repr__(self) -> str:
        return 'ModuleStub(%s, %s, %s)' % (
            tuple(self.function_stubs.values()), tuple(self.class_stubs.values()), repr(self.imports_stub))

class ImportBlockStub(Stub):
    def __init__(self, imports: ImportMap = None) -> None:
        self.imports = imports if imports else ImportMap()

    def render(self) -> str:
        imports = []
        for module in sorted(self.imports.keys()):
            names = sorted(self.imports[module])
            if module == '_io':
                module = module[1:]
            if len(names) == 1:
                imports.append("from %s import %s" % (module, names[0]))
            else:
                stanza = ["from %s import (" % (module,)]
                stanza.extend(["    %s," % (name,) for name in names])
                stanza.append(")")
                imports.append("\n".join(stanza))
        return "\n".join(imports)

```
Am I on the right track? I believe the order in both sequnces should be the same. The possible mismatch when both sequences are in order is that one sequence is shorter than the other. But I think this should not be possible because they are made from the same traces and the same traces should involve the same functions/classes/import. Please correct me if I oversimplied it.